pyproject.toml

```diff
@@ -58,6 +58,7 @@ imagecodecs = { version = ">=2024.1.1", optional = true }
 pyav = ">=12.0.5"
 moviepy = ">=1.0.3"
 rerun-sdk = ">=0.15.1"
+deepdiff = ">=7.0.1"
```

Comment:
TODO: move to another draft PR

Reply:
I still use this to warn the user if they try to provide a different config with their training command. It could protect against unexpected behaviour for users of other training frameworks where the behaviour is different.